src/Raven.Studio/typescript/viewmodels/database/indexes/deleteIndexesConfirm.ts

```diff
@@ -3,33 +3,75 @@ import dialog = require("plugins/dialog");
 import database = require("models/resources/database");
 import dialogViewModelBase = require("viewmodels/dialogViewModelBase");
 import messagePublisher = require("common/messagePublisher");
+import index = require("models/database/index/index");
+import indexDefinition = require("models/database/index/indexDefinition");
 
-class deleteIndexesConfirm extends dialogViewModelBase {
+class indexInfoForDelete {
+    indexName: string;
+    reduceOutputCollection: string;
+    referenceCollection: string;
 
-    deleteTask = $.Deferred<boolean>();
-    title: string;
+    readonly referenceCollectionExtension = "/References";
+    
+    static fromIndex(index: index) {
+        return new indexInfoForDelete(index.name, index.reduceOutputCollection(), !!index.reduceOutputReferencePattern());   
+    }
+    
+    static fromIndexDefinition(index: indexDefinition) {
+        return new indexInfoForDelete(index.name(), index.reduceToCollectionName(), index.patternForOutput());
```

Comment:
2 things here: 
Why in index we have: reduceOutputCollection but in index definiton we have: reduceToCollectionName?

Can we use same name? 

Same for:reduceToCollectionName vs patternForOutput. 

put `!!` before: index.patternForOutput())

Reply:
done - I unified the relevant variables in _index.ts_ and _indexDefinition.ts_